// graphql@16.9.0/utilities/index.d_20241210140727.ts
export { getIntrospectionQuery } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/getIntrospectionQuery';
export type {
  IntrospectionOptions,
  IntrospectionQuery,
  IntrospectionSchema,
  IntrospectionType,
  IntrospectionInputType,
  IntrospectionOutputType,
  IntrospectionScalarType,
  IntrospectionObjectType,
  IntrospectionInterfaceType,
  IntrospectionUnionType,
  IntrospectionEnumType,
  IntrospectionInputObjectType,
  IntrospectionTypeRef,
  IntrospectionInputTypeRef,
  IntrospectionOutputTypeRef,
  IntrospectionNamedTypeRef,
  IntrospectionListTypeRef,
  IntrospectionNonNullTypeRef,
  IntrospectionField,
  IntrospectionInputValue,
  IntrospectionEnumValue,
  IntrospectionDirective,
} from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/getIntrospectionQuery';
export { getOperationAST } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/getOperationAST';
export { getOperationRootType } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/getOperationRootType';
export { introspectionFromSchema } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/introspectionFromSchema';
export { buildClientSchema } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/buildClientSchema';
export { buildASTSchema, buildSchema } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/buildASTSchema';
export type { BuildSchemaOptions } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/buildASTSchema';
export { extendSchema } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/extendSchema';
export { lexicographicSortSchema } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/lexicographicSortSchema';
export {
  printSchema,
  printType,
  printIntrospectionSchema,
} from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/printSchema';
export { typeFromAST } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/typeFromAST';
export { valueFromAST } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/valueFromAST';
export { valueFromASTUntyped } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/valueFromASTUntyped';
export { astFromValue } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/astFromValue';
export { TypeInfo, visitWithTypeInfo } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/TypeInfo';
export { coerceInputValue } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/coerceInputValue';
export { concatAST } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/concatAST';
export { separateOperations } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/separateOperations';
export { stripIgnoredCharacters } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/stripIgnoredCharacters';
export {
  isEqualType,
  isTypeSubTypeOf,
  doTypesOverlap,
} from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/typeComparators';
export { assertValidName, isValidNameError } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/assertValidName';
export {
  BreakingChangeType,
  DangerousChangeType,
  findBreakingChanges,
  findDangerousChanges,
} from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/findBreakingChanges';
export type { BreakingChange, DangerousChange } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/findBreakingChanges';
export type { TypedQueryDocumentNode } from '.pnpm/graphql@16.9.0/node_modules/graphql/utilities/typedQueryDocumentNode';
