// @mysten/sui@1.16.1/dist/cjs/transactions/Commands.d_20241210140727.ts
import type { InferInput } from 'valibot';
import { Argument } from '.pnpm/@mysten+sui@1.16.1_typescript@5.7.2/node_modules/@mysten/sui/dist/cjs/transactions/data/internal.js';
import type { CallArg, Command } from '.pnpm/@mysten+sui@1.16.1_typescript@5.7.2/node_modules/@mysten/sui/dist/cjs/transactions/data/internal.js';
import type { Transaction } from '.pnpm/@mysten+sui@1.16.1_typescript@5.7.2/node_modules/@mysten/sui/dist/cjs/transactions/Transaction.js';
export type TransactionArgument = InferInput<typeof Argument> | ((tx: Transaction) => InferInput<typeof Argument>);
export type TransactionInput = CallArg;
export declare enum UpgradePolicy {
    COMPATIBLE = 0,
    ADDITIVE = 128,
    DEP_ONLY = 192
}
type TransactionShape<T extends Command['$kind']> = {
    $kind: T;
} & {
    [K in T]: Extract<Command, {
        [K in T]: any;
    }>[T];
};
/**
 * Simple helpers used to construct transactions:
 */
export declare const Commands: {
    MoveCall(input: {
        package: string;
        module: string;
        function: string;
        arguments?: Argument[];
        typeArguments?: string[];
    } | {
        target: string;
        arguments?: Argument[];
        typeArguments?: string[];
    }): TransactionShape<"MoveCall">;
    TransferObjects(objects: InferInput<typeof Argument>[], address: InferInput<typeof Argument>): TransactionShape<"TransferObjects">;
    SplitCoins(coin: InferInput<typeof Argument>, amounts: InferInput<typeof Argument>[]): TransactionShape<"SplitCoins">;
    MergeCoins(destination: InferInput<typeof Argument>, sources: InferInput<typeof Argument>[]): TransactionShape<"MergeCoins">;
    Publish({ modules, dependencies, }: {
        modules: number[][] | string[];
        dependencies: string[];
    }): TransactionShape<"Publish">;
    Upgrade({ modules, dependencies, package: packageId, ticket, }: {
        modules: number[][] | string[];
        dependencies: string[];
        package: string;
        ticket: InferInput<typeof Argument>;
    }): TransactionShape<"Upgrade">;
    MakeMoveVec({ type, elements, }: {
        type?: string;
        elements: InferInput<typeof Argument>[];
    }): TransactionShape<"MakeMoveVec">;
    Intent({ name, inputs, data, }: {
        name: string;
        inputs?: Record<string, InferInput<typeof Argument> | InferInput<typeof Argument>[]>;
        data?: Record<string, unknown>;
    }): TransactionShape<"$Intent">;
};
export {};
