// @mysten/sui@1.16.1/dist/cjs/transactions/executor/serial.d_20241210140727.ts
import { bcs } from '.pnpm/@mysten+sui@1.16.1_typescript@5.7.2/node_modules/@mysten/sui/dist/cjs/bcs/index.js';
import type { SuiClient, SuiTransactionBlockResponseOptions } from '.pnpm/@mysten+sui@1.16.1_typescript@5.7.2/node_modules/@mysten/sui/dist/cjs/client/index.js';
import type { Signer } from '.pnpm/@mysten+sui@1.16.1_typescript@5.7.2/node_modules/@mysten/sui/dist/cjs/cryptography/keypair.js';
import type { ObjectCacheOptions } from '.pnpm/@mysten+sui@1.16.1_typescript@5.7.2/node_modules/@mysten/sui/dist/cjs/transactions/ObjectCache.js';
import { Transaction } from '.pnpm/@mysten+sui@1.16.1_typescript@5.7.2/node_modules/@mysten/sui/dist/cjs/transactions/Transaction.js';
export declare class SerialTransactionExecutor {
    #private;
    constructor({ signer, defaultGasBudget, ...options }: Omit<ObjectCacheOptions, 'address'> & {
        client: SuiClient;
        signer: Signer;
        /** The gasBudget to use if the transaction has not defined it's own gasBudget, defaults to `50_000_000n` */
        defaultGasBudget?: bigint;
    });
    applyEffects(effects: typeof bcs.TransactionEffects.$inferType): Promise<[void, void]>;
    buildTransaction(transaction: Transaction): Promise<Uint8Array>;
    resetCache(): Promise<void>;
    waitForLastTransaction(): Promise<void>;
    executeTransaction(transaction: Transaction | Uint8Array, options?: SuiTransactionBlockResponseOptions): Promise<{
        digest: string;
        effects: string;
        data: import(".pnpm/@mysten+sui@1.16.1_typescript@5.7.2/node_modules/@mysten/sui/dist/cjs/client/index.js").SuiTransactionBlockResponse;
    }>;
}
export declare function getGasCoinFromEffects(effects: typeof bcs.TransactionEffects.$inferType): {
    ref: {
        objectId: string;
        digest: string;
        version: string;
    };
    owner: string;
};
